File: EGovernment_backend/police-service/domain/delict.go
package domain

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"
)

type Delict struct {
	ID                         primitive.ObjectID `bson:"_id" json:"id"`
	PolicemanID                string             `bson:"policeman_id" json:"policeman_id"`
	DriverIdentificationNumber string             `bson:"driver_identification_number" json:"driver_identification_number"`
	VehicleLicenceNumber       string             `bson:"vehicle_licence_number" json:"vehicle_licence_number"`
	DriverEmail                string             `bson:"driver_email" json:"driver_email"`
	Date                       primitive.DateTime `bson:"date" json:"date"`
	Location                   string             `bson:"location" json:"location"`
	Description                string             `bson:"description" json:"description"`
	DelictType                 DelictType         `bson:"delict_type" json:"delict_type"`
	DelictStatus               DelictStatus       `bson:"delict_status" json:"delict_status"`
	PriceOfFine                float64            `bson:"price_of_fine" json:"price_of_fine"`
	NumberOfPenaltyPoints      int64              `bson:"number_of_penalty_points" json:"number_of_penalty_points"`
}
type DelictType string

const (
	Speeding                                  = "Prekoracenje brzine"
	DrivingUnderTheInfluence                  = "Voznja pod uticajem narkotika"
	DrivingUnderTheInfluenceOfAlcohol         = "Voznja pod uticajem alkohola"
	ImproperOvertaking                        = "Nepravilno preticanje"
	ImproperParking                           = "Nepravilno parkiranje"
	FailureTooComplyWithTrafficLightsAndSigns = "Nepostovanje saobracajnih znakova"
	ImproperUseOfSeatBeltsAndChildSeats       = "Nepravilna upotreba sigurnosnog pojasa i sedista za decu"
	UsingMobilePhoneWhileDriving              = "Koriscenje mobilnog telefona tokom voznje"
	ImproperUseOfMotorVehicle                 = "Nepravilna upotreba motornih vozila"
	Other                                     = "Drugo"
)

type DelictStatus string

const (
	FineAwarded = "Novcana kazna dodeljena"
	FinePaid    = "Novcana kazna isplacena"
	SentToCourt = "Poslato na sud"
)

type DelictCreate struct {
	PolicemanID                string             `bson:"policeman_id" json:"policeman_id"`
	DriverIdentificationNumber string             `bson:"driver_identification_number" json:"driver_identification_number"`
	VehicleLicenceNumber       string             `bson:"vehicle_licence_number" json:"vehicle_licence_number"`
	DriverEmail                string             `bson:"driver_email" json:"driver_email"`
	Date                       primitive.DateTime `bson:"date" json:"date"`
	Location                   string             `bson:"location" json:"location"`
	Description                string             `bson:"description" json:"description"`
	DelictType                 DelictType         `bson:"delict_type" json:"delict_type"`
	DelictStatus               DelictStatus       `bson:"delict_status" json:"delict_status"`
	PriceOfFine                float64            `bson:"price_of_fine" json:"price_of_fine"`
	NumberOfPenaltyPoints      int64              `bson:"number_of_penalty_points" json:"number_of_penalty_points"`
}

type Citizen struct {
	JMBG        string    `bson:"jmbg" json:"jmbg"`
	Name        string    `bson:"name" json:"name"`
	Lastname    string    `bson:"lastname" json:"lastname"`
	DateOfBirth time.Time `bson:"date_of_birth" json:"date_of_birth"`
	//Address     string    `bson:"address" json:"address"`
}

File: EGovernment_frontend/src/app/models/traffic-police/delict.ts
import { DelictType } from "./delictType";
import { DelictStatus } from "./delictStatus";

export class Delict {
  id: string;
  policeman_id: string;
  driver_identification_number: string;
  vehicle_licence_number: string;
  driver_email: string;
  driver_jmbg: string;
  date: string;
  location: string;
  description: string;
  delict_type: DelictType;
  delict_status: DelictStatus;
  price_of_fine: number;
  number_of_penalty_points: number;
 
 
  constructor(id: string,
    policeman_id: string, 
    driver_identification_number: string, 
    vehicle_licence_number: string, 
    driver_email: string, 
    driver_jmbg: string, 
    date: string, 
    location: string, 
    description: string, 
    delict_type: DelictType, 
    delict_status: DelictStatus, 
    price_of_fine: number, 
    number_of_penalty_points: number ) 
    {
    this.id = id;
    this.policeman_id = policeman_id;
    this.driver_identification_number = driver_identification_number;
    this.vehicle_licence_number = vehicle_licence_number;
    this.driver_email = driver_email;
    this.driver_jmbg = driver_jmbg;
    this.date = date;
    this.location = location;
    this.description = description;
    this.delict_type = delict_type;
    this.delict_status = delict_status;
    this.price_of_fine = price_of_fine;
    this.number_of_penalty_points = number_of_penalty_points;
  }
}